extensions/bundlephobia/src/index.tsx

```diff
@@ -0,0 +1,98 @@
+import { Icon, List } from "@raycast/api";
+import { useState } from "react";
+import { fetchSizeBundlephobia, fetchSuggestionsBunldephobia } from "./fetchUtils";
+import { NpmsFetchResponse, Links } from "./packagesResponse";
+import { PackageResultModel } from "./packageRepsonse";
+import { formatInformationOfPackage } from "./utils";
+
+export default function Command() {
+  const [results, setResults] = useState<NpmsFetchResponse>([]);
+  const [detailResult, setDetailResult] = useState<PackageResultModel>(null);
+  const [loading, setLoading] = useState<boolean>(false);
+  const [loadingDetail, setLoadingDetail] = useState<boolean>(false);
+  const [searchTerm, setSearchTerm] = useState<string>("");
+  const [links, setLinks] = useState<Links | null>(null);
+
+  const onSearchTextChange = async (text: string) => {
+    setLoading(true);
+    const response = await fetchSuggestionsBunldephobia(text);
```

Comment:
For both `onSearchTextChange` and `onSelectionChange` you have to abort the previous call when making a new one otherwise you might end up with race conditions and show the result that doesn't correspond to the query/selection. Have a look at https://github.com/raycast/extensions/blob/main/examples/api-examples/src/typeahead-search.tsx if you need an example

Reply:
I resolved the race condition, please take a took @mathieudutour 🙏